### PR TITLE
Add support for creating nuget package Microsoft.Net.RoslynDiagnostic…

### PR DESCRIPTION
**Pull request by mavasani**
…s for consumption in Roslyn.sln

Also moved the repro to newer CodeAnalysis nuget packages (which have a pinned Microsoft.Net.Compilers package) and updated all the project nuspecs to include Analyzer.Utilities.

### CODE REVIEW

File: src/AggregateNugetPackages/FxCop/NuGet/FxCopAnalyzers.NuGet.proj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (mavasani) [inline src/AggregateNugetPackages/FxCop/NuGet/FxCopAnalyzers.NuGet.proj]: The FxCop nuget files just moved to a sub-folder - contents are same as before here.

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs [hunk 2]
```diff
@@ -6,27 +6,27 @@ internal static class RoslynDiagnosticIds
     {
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs]: This file now has new mappings for Roslyn specific rules.

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs [hunk 3]
```diff
@@ -6,27 +6,27 @@ internal static class RoslynDiagnosticIds
...
         public const string DeclarePublicApiRuleId = "RS0016";
         public const string RemoveDeletedApiRuleId = "RS0017";
-        public const string DoNotCreateTasksWithoutTaskSchedulerRuleId = "RS0018";
+        // public const string DoNotCreateTasksWithoutTaskSchedulerRuleId = "RS0018";   // Now RS0018 => System.Threading.Tasks.Analyzers.DoNotCreateTasksWithoutPassingATaskSchedulerAnalyzer
         public const string SymbolDeclaredEventRuleId = "RS0019";
-        public const string DeadCodeRuleId = "RS0020";
-        public const string DeadCodeTriggerRuleId = "RS0021";
-        public const string ExposedNoninstantiableTypeRuleId = "RS0022";
-        public const string MissingSharedAttributeRuleId = "RS0023";
+        // public const string DeadCodeRuleId = "RS0020";                               // Now ???
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs]: @srivatsn Was the dead code analyzer deleted? I am unable to find it..

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-analysis @lgolding

**Comment by lgolding** [comment]
I understand what you've done mechanically. What I don't understand is the purpose of this new meta-package Microsoft.Net.RoslynDiagnostics, and how its purpose differs from that of Microsoft.CodeAnalysis.FxCopAnalyzers.

Also why "Microsoft.Net"? Why not "Microsoft.CodeAnalysis" or "Microsoft.DotNet"? Is there precedent for the "Net" component of the package name?

**Comment by mavasani** [comment]
@lgolding Sorry, I should have added more context.

> What I don't understand is the purpose of this new meta-package Microsoft.Net.RoslynDiagnostics, and how its purpose differs from that of Microsoft.CodeAnalysis.FxCopAnalyzers

Microsoft.CodeAnalysis.FxCopAnalyzers will include all analyzer packages that have at least one FXCop rule implementation, while Microsoft.Net.RoslynDiagnostics includes only those packages which have at least one rule which is enabled for Roslyn.sln. The latter also includes some extremely Roslyn.sln specific rules which will never be candidates for inclusion in Microsoft.CodeAnalysis.FxCopAnalyzers (for e.g. see DoNotAccessDiagnosticDescriptorRuleId)


> Also why "Microsoft.Net"? Why not "Microsoft.CodeAnalysis" or "Microsoft.DotNet"? 

Roslyn.sln already consumes https://www.nuget.org/packages/Microsoft.Net.RoslynDiagnostics/1.1.1-beta1-20150818-01 for Roslyn analyzers - this package was getting built out of Roslyn repo, prior to us moving all Roslyn analyzers to roslyn-analyzers repo.

**Comment by lgolding** [comment]
:+1:

**Pull request merged**